### ISSUE TITLE
Error on tasks "locked for mapping" filter

### ISSUE THREAD

**Issue by willemarcel** [hotosm/tasking-manager#2559]
When fetching https://tasking-manager-tm4-api.hotosm.org/api/v2/users/360183/tasks/?status=LOCKED_FOR_MAPPING

`
{"tasks": [{"taskId": 14, "projectId": 18, "taskStatus": "LOCKED_FOR_MAPPING", "lockHolder": "wille", "taskHistory": [], "taskAnnotation": null, "isUndoable": false, "autoUnlockSeconds": 7200.0, "lastUpdated": "2020-03-23T17:21:36.078521Z"}, {"taskId": 174, "projectId": 18, "taskStatus": "LOCKED_FOR_MAPPING", "lockHolder": "RAytoun", "taskHistory": [], "taskAnnotation": null, "isUndoable": false, "autoUnlockSeconds": 7200.0, "lastUpdated": "2020-03-23T16:45:45.849443Z"}], "pagination": {"hasNext": false, "hasPrev": false, "nextNum": null, "page": 1, "pages": 1, "prevNum": null, "perPage": 10, "total": 2}}
`

It's returning one task locked by me and another locked by RAytoun

**Comment by willemarcel** [comment]
![image](https://user-images.githubusercontent.com/666291/77346402-f4ad6800-6d14-11ea-9298-465e12215def.png)

**Comment by ramyaragupathy** [comment]
Looking at this - we're retrieving by the right user id from task history table.
* Wille has the task unlocked at `2020-03-23 16:45:45.849443`
* Ralph has the task auto unlocked at `2020-03-23 17:17:23.175186`

Wille's task history entry has been recorded prior to Ralph's. In our code, during auto-unlock, we only update the existing `LOCKED_FOR_MAPPING` entry and Ralph's unlock has happened after Wille's unlock. 
So was Wille allowed to lock the task that was locked by Ralph ? in which case Ralph's task history id should have been recorded earlier, but the entry comes only after Wille's action. Was the auto-unlock triggered in less than 2 hours?

@willemarcel - Also noticing that while looking at task history, other than `count-unread` for notifications there are repeated requests sent to:
* `latest` - 7 to 8 mins response time 😱 
* `openstreetmap/capabilites`

**Comment by ramyaragupathy** [comment]
@willemarcel - did you get any error message while unlocking? I see one error for `Attempting to unlock a task owned by another user` in the logs

**Comment by willemarcel** [comment]
@ramyaragupathy I haven't tried to unlock the task... initially I thought it was only a problem on that endpoint query (and also caused by how task history is stored, we talked about it some weeks ago...).

So could you check if there is something we could improve on the query? We can talk about it later

**Comment by ramyaragupathy** [comment]
Per chat: 

* `projects/tasks/queries/own/locked/` will be used to identify any tasks that have been locked by the user in the current instance
* `projects/tasks/queries/own/locked/details/` will be called if the above endpoint returns a list
* `projects/{project_id}/tasks?user={user_id}&status=locked` were supposed to be replacement for `projects/tasks/queries/own/locked/` and `projects/tasks/queries/own/locked/details/`. However now they serve a different purpose - which is to list user tasks from `My contributions` section.

**Comment by ramyaragupathy** [comment]
Fixed locked tasks related issues through https://github.com/hotosm/tasking-manager/pull/2562. Now looking at `my tasks` endpoint for `My Contributions` page

**Comment by ramyaragupathy** [comment]
* `https://tm4.hotosm.org/contributions/tasks` returns tasks based on their current status. Action buttons on this page do not refer to user activity on these tasks ⚠️ 

![image](https://user-images.githubusercontent.com/12103383/77437490-ba25f700-6e0a-11ea-80c5-88e0b4811774.png)

eg:
`Mapped`: Tasks touched by the user that are currently in the `MAPPED` state
`Validated`: Tasks touched by the user that are currently in the `VALIDATED` state
`Invalidated`: Tasks touched by the user that are currently in the `INVALIDATED` state

**Comment by ramyaragupathy** [comment]
With the above purpose in context, https://tasking-manager-tm4-api.hotosm.org/api/v2/users/360183/tasks/?status=LOCKED_FOR_MAPPING is right in returning two entries - all tasks touched by user id 360183 that are currently in LOCKED_FOR_MAPPING state. Hence this is not a bug.

@willemarcel - request you to use `projects/tasks/queries/own/locked/details/` and `projects/tasks/queries/own/locked/` to identify user's locked tasks.

cc @xamanu

**Comment by willemarcel** [comment]
With #2597, the only remaining issue is to fix the lock restrictions:

If I have a task locked for mapping or for validation, the api is allowing me to lock another task for validation. It's not the desired behaviour. We can lock multiple tasks for validation, but they need to be on the same project and locked on the same API request.

The lock for mapping restrictions are working correctly.
